suppressible-error-prone/src/main/java/com/palantir/suppressibleerrorprone/SuppressWarningsCoalesce.java

```diff
@@ -0,0 +1,160 @@
+        summary = "blah")
+public final class SuppressWarningsCoalesce extends BugChecker
+        implements BugChecker.MethodTreeMatcher, BugChecker.VariableTreeMatcher, BugChecker.ClassTreeMatcher {
+
+    private static final MultiMatcher<Tree, AnnotationTree> HAS_REPEATABLE_SUPPRESSION = Matchers.annotations(
+            MatchType.AT_LEAST_ONE, Matchers.isType("com.palantir.suppressibleerrorprone.RepeatableSuppressWarnings"));
+
+    @Override
+    public Description matchMethod(MethodTree tree, VisitorState state) {
+        return coalesceSuppressWarnings(state, tree, tree.getModifiers());
+    }
+
+    @Override
+    public Description matchVariable(VariableTree tree, VisitorState state) {
+        return coalesceSuppressWarnings(state, tree, tree.getModifiers());
+    }
+
+    @Override
+    public Description matchClass(ClassTree tree, VisitorState state) {
+        return coalesceSuppressWarnings(state, tree, tree.getModifiers());
+    }
+
+    private Description coalesceSuppressWarnings(VisitorState state, Tree tree, ModifiersTree modifiersTree) {
+        if (!HAS_REPEATABLE_SUPPRESSION.matches(tree, state)) {
+            return Description.NO_MATCH;
+        }
+
+        List<? extends AnnotationTree> suppressWarnings = modifiersTree.getAnnotations().stream()
+                .filter(annotation -> {
+                    Name annotationName = annotationName(annotation.getAnnotationType());
+                    return annotationName.contentEquals("SuppressWarnings")
+                            || annotationName.contentEquals("RepeatableSuppressWarnings");
+                })
+                .collect(Collectors.toList());
+
+        if (suppressWarnings.isEmpty()) {
+            return Description.NO_MATCH;
+        }
+
+        List<String> warningsToSuppress = suppressWarnings.stream()
+                .sorted(suppressWarningsBeforeRepeatableSuppressedWarnings())
+                .flatMap(SuppressWarningsCoalesce::annotationStringValues)
+                .collect(Collectors.toList());
+
+        if (warningsToSuppress.isEmpty()) {
+            return Description.NO_MATCH;
+        }
+
+        SuggestedFix.Builder fixBuilder = SuggestedFix.builder();
+        suppressWarnings.forEach(fixBuilder::delete);
+
+        String suppressWarningsString = '"' + String.join("\", \"", warningsToSuppress) + '"';
+
+        if (warningsToSuppress.size() > 1) {
+            suppressWarningsString = "{" + suppressWarningsString + "}";
+        }
+
+        fixBuilder.prefixWith(tree, "@SuppressWarnings(" + suppressWarningsString + ")");
+
+        return buildDescription(tree)
+                .setMessage("Coalescing @SuppressWarnings annotations")
+                .addFix(fixBuilder.build())
+                .build();
+    }
+
+    private static Name annotationName(Tree annotationType) {
+        if (annotationType instanceof IdentifierTree) {
+            return ((IdentifierTree) annotationType).getName();
+        }
```

Comment:
you should be able to add a variable and refer to that without having to cast anymore!

```java
if (annotaitonType instanceof IdentifierTree identifierTree) {
  return identifierTree.getName();
}
```

etc.

You could even do it via a switch if you were so inclined:

```java
switch (annotationType) {
  case IdentifierTree identifierTree -> identifierTree.getName();
  case MemberSelectTree memberSelectTree -> memberSelectTree.getIdentifier();
  default -> throw new UnsupportedOperationException(...);
```

Reply:
actually, switch pattern matching is only java 21, ignore me